### PR TITLE
fix: logger method and refactoring little things

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [X] Bugfix
- [ ] New feature
- [ ] Enhancement
- [X] Refactoring

## :scroll: Description
fix: logger method and refactoring little things

## :bulb: Motivation and Context
if `debug` was enabled, it'd hang on mobile, but it'd work on web.
integrations could be triggered before `hub` being fully enabled.
making unnecessary methods and classes as private.

## :green_heart: How did you test it?
running it

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [X] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: dart/lib/src/sentry_client.dart [hunk 1]
```diff
@@ -11,19 +11,24 @@ import 'version.dart';
...
 
-  Random _random;
+  final Random _random;
 
   static final _sentryId = Future.value(SentryId.empty());
 
+  /// Instantiates a client using [SentryOptions]
+  SentryClient._(SentryOptions options)
+      : _options = options,
+        _random = options.sampleRate == null ? null : Random();
```
Comment (rxlabz) [inline dart/lib/src/sentry_client.dart]:
```suggestion
  SentryClient._(this._options)
      : _random = _options.sampleRate == null ? null : Random();
```

File: dart/lib/src/sentry_client.dart [hunk 2]
```diff
@@ -11,19 +11,24 @@ import 'version.dart';
...
   }
 
   final SentryOptions _options;
 
-  Random _random;
+  final Random _random;
 
   static final _sentryId = Future.value(SentryId.empty());
 
+  /// Instantiates a client using [SentryOptions]
```
Comment (rxlabz) [inline dart/lib/src/sentry_client.dart:27]: Why did you add a private ctor ?

File: dart/lib/src/transport/http_transport.dart [hunk 3]
```diff
@@ -16,18 +17,26 @@ class HttpTransport implements Transport {
...
+    if (options.httpClient is NoOpClient) {
+      options.httpClient = Client();
+    }
+
+    return HttpTransport._(options);
+  }
+
+  HttpTransport._(SentryOptions options)
       : _options = options,
         _dsn = Dsn.parse(options.dsn),
```
Comment (rxlabz) [inline dart/lib/src/transport/http_transport.dart]:
```suggestion
  HttpTransport._(this._options)
      : _dsn = Dsn.parse(_options.dsn),
```

File: dart/lib/src/sentry_client.dart (hunk 2 above)
Comment (marandaneto) [reply dart/lib/src/sentry_client.dart:27]: because we have a public factory, so we can separate a bit more the things

**Review by rxlabz — COMMENTED** [review]
LGTM

**Review by rxlabz — APPROVED** [review]
LGTM
